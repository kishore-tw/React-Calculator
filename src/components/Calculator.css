.formBox {
    width: 300px;
    height: 350px;
    border-radius:20px;
    background-color: gray;
    padding : 60px ;
    display: block;
    margin-left: auto;
    margin-right: auto
}

.button{
    padding-right: 5px;
    padding-top: 150px;
}

input,
input::-webkit-input-placeholder {
    font-size: 13px;
    line-height: 2;
}

input[type=number]::-webkit-inner-spin-button, 
input[type=number]::-webkit-outer-spin-button { 
  -webkit-appearance: none; 
  margin: 0; 
}

.input{
    background-color: white;
    height:35px;
    padding-left: 12px;
    font-size:0.2px;
    width: 300px;
}
