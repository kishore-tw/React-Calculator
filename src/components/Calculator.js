import React from 'react'
import './Calculator.css'
import Grid from '@material-ui/core/Grid';
import { Button, TextField, Input } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

function Calculator({inputOne, inputTwo, output, hasError, onValueChange, onClickChange}){
return (
    <form>
    <Grid container spacing={2} direction="column">
    <Grid className="formBox">
        <Grid>
        { hasError === true &&
        <Alert severity="error">Please fill both the fields</Alert>
        }
        { hasError === 'validation_error' &&
        <Alert severity="error">Please enter valid values</Alert>
        }
        <Grid><br/></Grid>
        </Grid>
        <Grid  item><Input value={inputOne} onChange = {data => onValueChange("inputOne", data.target.value)}
        className="input" id = "input_one"  placeholder="Input One"/>
        </Grid>
      
        <Grid><br/></Grid>
        <Grid item><Input value={inputTwo} onChange = {data => onValueChange("inputTwo", data.target.value)}
        className="input" id = "input_two" placeholder="Input Two"/></Grid>
        <Grid><br/></Grid>

        <Grid item>
        <span className="button"><Button  id="plus" onClick = {data => onClickChange("+")} variant="contained" color="default"> + </Button></span>
        <span className="button"><Button  id="minus" onClick = {data => onClickChange("-")} variant="contained" color="default"> - </Button></span>
        <span className="button"><Button  id="multiply" onClick = {data => onClickChange("*")} variant="contained" color="default"> * </Button></span>
        <span className="button"><Button  id="divide" onClick = {data => onClickChange("/")} variant="contained" color="default"> / </Button></span>
        </Grid>

        <Grid><br/><br/></Grid>
        <Grid item>
            <Button variant="contained" id="clear" onClick = {data => onClickChange("clear")}> Clear </Button></Grid>
        <Grid><br/></Grid>
        <Grid><br/></Grid>
        <TextField className = "textField" size="small" label="Output" variant="outlined" id="output" value={output}  type="number" 
        InputProps={{
            readOnly: true,
          }}/>
    </Grid>
    </Grid>
    </form>
   )
}

export default Calculator