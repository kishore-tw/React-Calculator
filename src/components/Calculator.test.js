import React from "react";
import renderer from 'react-test-renderer';
import Calculator from './Calculator'

describe(' test suite for Calculator', () => {
    test('should render calculator component for given parameters', () => {
        const mockInputOne = 2;
        const mockInputTwo = 3;
        const mockOutput = 5;
        const mockHasError = false;
        const mockCalculator = renderer.create(<Calculator inputOne={mockInputOne} inputTwo = {mockInputTwo} output = {mockOutput} 
            hasError={mockHasError} />);
        expect(mockCalculator).toMatchSnapshot();
    })
});


