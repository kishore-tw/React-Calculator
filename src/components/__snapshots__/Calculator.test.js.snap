// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test suite for Calculator should render calculator component for given parameters 1`] = `
<form>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
  >
    <div
      className="MuiGrid-root formBox"
    >
      <div
        className="MuiGrid-root"
      >
        <div
          className="MuiGrid-root"
        >
          <br />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline input"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            id="input_one"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Input One"
            type="text"
            value={2}
          />
        </div>
      </div>
      <div
        className="MuiGrid-root"
      >
        <br />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline input"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            id="input_two"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Input Two"
            type="text"
            value={3}
          />
        </div>
      </div>
      <div
        className="MuiGrid-root"
      >
        <br />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <span
          className="button"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            id="plus"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
               + 
            </span>
          </button>
        </span>
        <span
          className="button"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            id="minus"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
               - 
            </span>
          </button>
        </span>
        <span
          className="button"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            id="multiply"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
               * 
            </span>
          </button>
        </span>
        <span
          className="button"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            id="divide"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
               / 
            </span>
          </button>
        </span>
      </div>
      <div
        className="MuiGrid-root"
      >
        <br />
        <br />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained"
          disabled={false}
          id="clear"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
             Clear 
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root"
      >
        <br />
      </div>
      <div
        className="MuiGrid-root"
      >
        <br />
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root textField"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink={true}
          htmlFor="output"
          id="output-label"
        >
          Output
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="output"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="number"
            value={5}
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
            >
              <span>
                Output
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</form>
`;
